linkerd/distribute/src/cache.rs

```diff
@@ -0,0 +1,73 @@
+use super::{params, NewDistribute};
+use linkerd_stack::{layer, NewService, Param};
+use parking_lot::Mutex;
+use std::{fmt::Debug, hash::Hash, sync::Arc};
+
+/// A [`NewService`] that produces [`NewDistribute`]s using a shared cache of
+/// backends.
+#[derive(Debug)]
+pub struct CacheNewDistribute<K, N, S>(Arc<Inner<K, N, S>>);
+
+#[derive(Debug)]
+pub struct Inner<K, N, S> {
```

Comment:
AFAICT this doesn't actually need to be `pub`? and it looks like a private implementation detail...
```suggestion
struct Inner<K, N, S> {
```